packages/netlify-cms-core/src/components/Editor/EditorControlPane/Widget.js

```diff
@@ -90,7 +90,11 @@ export default class Widget extends Component {
   validate = (skipWrapped = false) => {
     const { field, value } = this.props;
     const errors = [];
-    const validations = [this.validatePresence, this.validatePattern];
+    const validations =
+      field.get('widget') === 'number'
```

Comment:
The CMS core can't know about specific widgets. We either need to include this in a way that's widget agnostic, or else move this to the `validate` method in the Number widget control component. I'm inclined to say the latter.

What do you think?

Reply:
> As a final note, I considered simply implementing an `isValid()` method inside `NumberControl` as it would be already in the loop for validation out of the box (not requiring the changes in `EditorControlPane/Widget.js`) but for the **error strings to be translatable** we'd have to always provide the base **in the `defaultPhrases` constants** as the Number widget is part of the core. The **error type would also be `CUSTOM` instead of the new `RANGE`** type, which shouldn't matter as much but still... Let me know what you think.

@erquhart If you're happy with the caveats presented (phrases and error type, basically), I agree completely with moving this to the scope of `NumberControl`. I think, ideally, the core widgets should be treated in the same way as custom widgets.

As you mentioned, we could also validate for `min` and `max` options on all widgets (same as presence and pattern) but such validations will always be different from widget to widget so we'd have to implement different validators for all the core widgets on which this could make any sense plus we'd need to ask for custom validators in custom widgets... All in all, not the best option IMHO.

Reply:
Totally missed the final note at the bottom of your OP, apologies for that. I think we can put up with the caveats for now, it'll apply more pressure for us to improve the Widget API, which needs to happen sooner than later.

Reply:
Roger, pushing the new code